lib/bq_stream/bq_stream.rb

```diff
@@ -80,7 +80,8 @@ def dequeue_items
       OldestRecord.update_bq_earliest
       create_bq_writer
       # records = QueuedItem.all.limit(batch_size) # removing for testing TODO: update after test
-      records = QueuedItem.where('id > ?', start_after_id).limit(batch_size)
+      records = QueuedItem.where('id < ?', start_after_id).where(sent_to_bq: nil)
+      records += QueuedItem.where('id > ?', start_after_id).limit(batch_size - records.count)
```

Comment:
Why always search by the last sent_to_bq? Why not just do `QueuedItem.where(send_to_bq: nil).limit(batch_size)` ? If there's no missed records, you'll start where you want to anyways. If there are missed records, then you'll get those  too.